wgpu-core/src/command/mod.rs

```diff
@@ -80,30 +80,33 @@ pub(crate) enum CommandEncoderStatus {
     /// [`Global::command_encoder_finish`] transitions a
     /// `CommandBuffer` from the `Recording` state into this state.
     ///
-    /// [`Global::queue_submit`] drops command buffers unless they are
+    /// [`Global::queue_submit`] requires that command buffers are
     /// in this state.
+    ///
+    /// This corresponds to WebGPU's "ended" state.
+    /// See <https://www.w3.org/TR/webgpu/#encoder-state-ended>
     Finished(CommandBufferMutable),
 
-    /// An error occurred while recording a compute or render pass.
+    /// The command encoder is invalid.
     ///
-    /// When a `CommandEncoder` is left in this state, we have also
-    /// returned an error result from the function that encountered
-    /// the problem. Future attempts to use the encoder (for example,
-    /// calls to [`Self::record`]) will also return errors.
-    Error,
+    /// The error that caused the invalidation is stored here, and will
+    /// be raised by `CommandEncoder.finish()`.
+    Error(CommandEncoderError),
+
+    /// Temporary state used by the functions below. Encoder should never be
```

Comment:
thought: "below" should still work (AFAIK this will always be above all methods in `rustdoc`'s output), it's just odd to see in documentation commentary. 🤔